### ISSUE TITLE
model.get_layer fails to work if called a second time after adding a new layer

### ISSUE THREAD

**Issue by bruceweir** [fchollet/keras#7308]
model.get_layer() does not correctly return the correct layer if a new layer is added after it is first used.

To replicate (using keras 2.0.6)

from keras.models import Sequential
from keras.layers import Dense, Dropout, Conv2D, Flatten, Reshape

model = Sequential()
model.add(Dense(128, input_shape=(784,)))
model.add(Dense(64))
model.get_layer(index=-1).output_shape
model.add(Dense(32))
model.get_layer(index=-1).output_shape

----- outputs: (None, 64), rather than the expected (None, 32)

**Comment by Dref360** [comment]
Good catch!
The issue happens because the model is not rebuilt during the second call.
The fix is certainly not to rebuild the model each time. 
Maybe we could improve the doc to specify that behavior.

**Comment by bruceweir** [comment]
Being able to access the calculated output shape of a previous layer is handy for building networks programmatically. Is there an alternative method for getting that information?

**Comment by Dref360** [comment]
There is a PR to fix it.
